test/apps/iodemo/io_demo.cc

```diff
@@ -302,22 +339,36 @@ class P2pDemoCommon : public UcxContext {
             return tr_hdr()->sn;
         }
 
-        uint16_t chksum() const {
-            return chk_hdr()->chk_sum;
+        uint32_t seed() const {
+            return chk_hdr()->seed;
```

Comment:
maybe we don't need to validate seed according to header and pass it in chk_hdr?
IO_READ: sender generate a seed in io_message, and expect the returned data to match that seed
IO_WRITE: sender generate a seed in io_message, generate data by that send. receiver will expect the arrived data to match that seed

Reply:
this is the change in data integrity check algo, now every message is checked separately (including io_message since it can be sent in fragments) and transaction is checked by `sn`. you proposal assumes that IO data receiver should have some storage of outstanding transactions and do matching and validation on transaction completion.
if you think that this approach is better, let's do it in another PR?

Reply:
done